source/treap.h

```diff
@@ -15,7 +15,7 @@ struct Treap {
 
 	Treap(Val x, Treap* l = NULL, Treap *r = NULL) {
 		val = x; sz = 1; prio = rng();
-		left = l; right = r; rev = false; sum = 0;
+		left = l; right = r; rev = false; sum = x;
```

Comment:
Can we just put this->recalc(); here? Because this doesn't handle the case when left and right are non null

Reply:
You are right.